api_tests/lib/ledgers/lnd_instance.ts

```diff
@@ -0,0 +1,184 @@
+
+        this.logger.debug("Waiting for lnd password RPC server");
+        await this.logReader().waitForLogMessage(
+            "RPCS: password RPC server listening"
+        );
+
+        {
+            const config = {
+                server: this.getGrpcSocket(),
+                tls: this.tlsCertPath(),
+            };
+            this.logger.debug("Instantiating lnd connection:", config);
+            const lnd = await Lnd.init(config);
+
+            const { cipherSeedMnemonic } = await lnd.lnrpc.genSeed({});
+            const walletPassword = Buffer.from("password", "utf8");
+            await lnd.lnrpc.initWallet({ cipherSeedMnemonic, walletPassword });
+        }
+
+        this.logger.debug("Waiting for lnd unlocked RPC server");
+        await this.logReader().waitForLogMessage("RPCS: RPC server listening");
+        this.logger.debug(
+            "Waiting for admin macaroon file to exist:",
+            this.adminMacaroonPath()
+        );
+        await waitUntilFileExists(this.adminMacaroonPath());
+
+        this.logger.debug("Waiting for lnd to catch up with blocks");
+        await this.logReader().waitForLogMessage(
+            "LNWL: Done catching up block hashes"
+        );
+
+        const config = {
+            server: this.getGrpcSocket(),
+            tls: this.tlsCertPath(),
+            macaroonPath: this.adminMacaroonPath(),
+        };
+        this.logger.debug("Instantiating lnd connection:", config);
+        this.lnd = await Lnd.init(config);
+
+        this.publicKey = (await this.lnd.lnrpc.getInfo()).identityPubkey;
+        this.logger.info("lnd is ready:", this.publicKey);
+    }
+
+    public stop() {
+        this.logger.debug("Stopping lnd instance");
+        this.process.kill("SIGTERM");
+        this.process = null;
+    }
+
+    public isRunning() {
+        return this.process != null;
+    }
+
+    public logPath() {
+        return path.join(this.lndDir, "logs", "bitcoin", "regtest", "lnd.log");
+    }
+
+    public tlsCertPath() {
+        return path.join(this.lndDir, "tls.cert");
+    }
+
+    public adminMacaroonPath() {
+        return path.join(
+            this.lndDir,
+            "data",
+            "chain",
+            "bitcoin",
+            "regtest",
+            "admin.macaroon"
+        );
+    }
+
+    public getGrpcSocket() {
+        return `${this.getGrpcHost()}:${this.getGrpcPort()}`;
+    }
+
+    public getGrpcHost() {
+        return "127.0.0.1";
+    }
+
+    public getGrpcPort() {
+        return this.actorConfig.lndRpcPort;
+    }
+
+    public getLightningSocket() {
+        return `${this.getLightningHost()}:${this.getLightningPort()}`;
+    }
+
+    public getLightningHost() {
+        return "127.0.0.1";
+    }
+
+    public getLightningPort() {
+        return this.actorConfig.lndP2pPort;
+    }
+
+    private async createConfigFile(lndDir: string) {
+        // We don't use REST but want a random port so we don't get used port errors.
```

Comment:
You could disable the interface as-well? :)

Reply:
We just wrote code in cnd to hit lnd via the REST endpoint.  We have a mis-match here, I assume we are going to have e2e tests eventually that hit the REST endpoint also in order to mimic things we do in cnd?

Reply:
No, we unfortunately cannot ☹️

Reply:
Fair enough yes, what was I thinking :D